R/tar-stars.R

```diff
@@ -0,0 +1,136 @@
+                      pattern = NULL,
+                      driver = NULL,
+                      options = NULL,
+                      ...,
+                      tidy_eval = targets::tar_option_get("tidy_eval"),
+                      packages = targets::tar_option_get("packages"),
+                      library = targets::tar_option_get("library"),
+                      repository = targets::tar_option_get("repository"),
+                      iteration = targets::tar_option_get("iteration"),
+                      error = targets::tar_option_get("error"),
+                      memory = targets::tar_option_get("memory"),
+                      garbage_collection = targets::tar_option_get("garbage_collection"),
+                      deployment = targets::tar_option_get("deployment"),
+                      priority = targets::tar_option_get("priority"),
+                      resources = targets::tar_option_get("resources"),
+                      storage = targets::tar_option_get("storage"),
+                      retrieval = targets::tar_option_get("retrieval"),
+                      cue = targets::tar_option_get("cue")) {
+
+    name <- targets::tar_deparse_language(substitute(name))
+
+    envir <- targets::tar_option_get("envir")
+
+    command <- targets::tar_tidy_eval(
+        expr = as.expression(substitute(command)),
+        envir = envir,
+        tidy_eval = tidy_eval
+    )
+
+    pattern <- targets::tar_tidy_eval(
+        expr = as.expression(substitute(pattern)),
+        envir = envir,
+        tidy_eval = tidy_eval
+    )
+
+    # if not specified by user, pull the corresponding geotargets option
+    driver <- driver %||% geotargets_option_get("gdal.raster.driver")
+    options <- options %||% geotargets_option_get("gdal.raster.creation_options")
+
+    targets::tar_target_raw(
+        name = name,
+        command = command,
+        pattern = pattern,
+        packages = packages,
+        library = library,
+        format = create_format_stars(driver = driver, options = options, ...),
+        repository = repository,
+        iteration = iteration,
+        error = error,
+        memory = memory,
+        garbage_collection = garbage_collection,
+        deployment = deployment,
+        priority = priority,
+        resources = resources,
+        storage = storage,
+        retrieval = retrieval,
+        cue = cue
+    )
+}
+#' @param driver character. File format expressed as GDAL driver names passed to [stars::write_stars()]. See [sf::st_drivers()].
+#' @param options character. GDAL driver specific datasource creation options passed to [stars::write_stars()]
+#' @param ... Additional arguments not yet used
+#' @noRd
+create_format_stars <- function(driver, options, ...) {
+
+    if (!requireNamespace("sf")) {
+        stop("package 'sf' is required", call. = FALSE)
+    }
+
+    if (!requireNamespace("stars")) {
+        stop("package 'stars' is required", call. = FALSE)
+    }
```

Comment:
Maybe move this to the very top of `tar_stars()`

Reply:
will do to mirror your other changes/suggestions!

Reply:
done in https://github.com/njtierney/geotargets/pull/33/commits/0e20c172f574efed855625ec06219c40eade9917